txnsync/peerscheduler.go

```diff
@@ -55,9 +97,32 @@ func (p *peerScheduler) Len() int {
 	return len(p.peers)
 }
 
+func (p *peerScheduler) replaceIndices(indices []int, i, j int) {
+
+	for idx, x := range indices {
+		if x == i {
+			indices[idx] = j
+		} else if x == j {
+			indices[idx] = i
+		}
+	}
+	sort.Slice(indices, func(i, j int) bool {
+		return p.peers[indices[i]].next < p.peers[indices[j]].next
+	})
+}
+
 // Swap implements heap.Interface
 func (p *peerScheduler) Swap(i, j int) {
 	p.peers[i], p.peers[j] = p.peers[j], p.peers[i]
+	if p.peers[i].peer == p.peers[j].peer {
+		indices := p.nextPeers[p.peers[i].peer]
+		sort.Slice(indices, func(i, j int) bool {
+			return p.peers[indices[i]].next < p.peers[indices[j]].next
+		})
```

Comment:
consider avoiding using the same variable names are above ( i.e. `i` and `j` ); pick different variable names.